spec/gemstash/http_client_spec.rb

```diff
@@ -20,6 +20,16 @@
     @other_server.stop
   end
 
+  describe "#for" do
+    it "sets configured timeout value" do
+      configured_timeout = 99
+      client = Gemstash::HTTPClient.for(Gemstash::Upstream.new(@server.url), configured_timeout)
+      expect(
+        client.instance_variable_get(:@client).options.timeout
+      ).to eq(configured_timeout)
+    end
```

Comment:
PS: I pushed a sample commit that does what I describe above. It isn't as elegant as I'd prefer but it tests behavior.